### DISCUSSION TITLE
does typespec support header on namespace/interface?

### DISCUSSION THREAD

**Discussion by sku0x20** [microsoft/typespec#5370] [Q&A]
a header which should be included in all api calls and they are invalid without it.  
eg. `App-Version`

**Comment by allenjzhang** [comment]
header is another property field that can only exist on models. Having said that, TypeSpec offers layered extensibility which you can further enhance/apply team/org level API patterns on top of lower level construct. For example, `rest` operations are built on lower `http` primitives. You can create custom operation/interface template to codify the inclusion of said header or others such as standard error response. To enforce the adherence of the org/team requirements, custom linter can be added if you publish these common patterns in a library.
